shell/src/main/java/org/apache/kafka/shell/MetadataShell.java

```diff
@@ -81,6 +85,21 @@ public MetadataShell build() {
         }
     }
 
+    static FileLock takeFileLock(File directory) {
+        FileLock fileLock = new FileLock(new File(directory, ".lock"));
```

Comment:
I suggest not creating a `.lock` file if the specified directory doesn't contain a `.lock` file.

Reply:
That seems like a bad idea since it could allow another server or tool to collide with what we're doing.